### PR TITLE
[PPOTrainer]PPO_PTX Support for mixed training loss (ppo reward loss + pretrained data loss)

### PR DESCRIPTION
**Pull request by 1485840691**
This PR is to to support ppo_ptx loss 
         loss = PPO_loss + coef * ptx_loss_from_pretrained_data
Requirement is from Issue: https://github.com/huggingface/trl/issues/1050

Done: 
   Step 1 : update ppo_trainer and hello_world_example

Done:
    Step 2: Update related real examples to integrate ppo_ptx

[ppo_ptx wandb report](https://wandb.ai/open_source/trl/reports/ppo_ptx-wandb-report-from-running-example--Vmlldzo2NjE3NzQ4)
image
image

[ppo wandb report](https://wandb.ai/open_source/trl/reports/ppo-example-wandb-results--Vmlldzo2NjI2MTU2)
image
image

### CODE REVIEW

File: trl/trainer/ppo_trainer.py [hunk 1]
```diff
@@ -1246,6 +1417,18 @@ def loss(
         )
         return pg_loss, self.config.vf_coef * vf_loss, flatten_dict(stats)
 
+    def compute_ptx_loss(self, ptx_model_inputs: Dict[str, Any]):
+        ptx_loss = None
+        if self.ptx_loss_args.ptx_coef > 0 and ptx_model_inputs:
+            ptx_outputs = self.model(**ptx_model_inputs)
+            if isinstance(ptx_outputs, tuple) or isinstance(ptx_outputs, list) and len(ptx_outputs) > 1:
```
Comment (1485840691-eng) [inline trl/trainer/ppo_trainer.py:1424]: There is a bug here, have fixed it.

**Comment by 1485840691** [comment]
@lvwerra @younesbelkada  could you please help review this PR? It has been for a while and would like to know what more to do next for this PR?
